packages/babel-core/src/transformation/file/options/option-manager.js

```diff
@@ -281,8 +281,18 @@ export default class OptionManager {
           val = require(presetLoc);
         }
 
-        // If the imported preset is a transpiled ES2015 module, grab the default export.
-        if (typeof val === "object" && val.__esModule) val = val.default;
+        // If the imported preset is a transpiled ES2015 module
+        if (typeof val === "object" && val.__esModule) {
+          // Try to grab the default export.
+          if (val.default) {
+            val = val.default;
+          } else {
+            // If there is no default export we treat all named exports as options
+            // and just remove the __esModule
+            const { __esModule:_, ...rest } = val;
```

Comment:
Hitting no-unused-vars with https://github.com/babel/babel-eslint/issues/95

  292:32  error  '_' is defined but never used  no-unused-vars